File: ios/MobilePassSDK/Service/DataService.swift
//
//  DataService.swift
//  MobilePassSDK
//
//  Created by Erinc Cakir on 17.02.2021.
//

import Foundation

class DataService {
    func sendUserInfo(request: RequestSetUserData, completion: @escaping (Result<ResponseDefault?, RequestError>) -> Void) {
        BaseService.shared.requestPost(url: "api/v1/setpublickey", data: request.dictionary, completion: completion)
    }
    
    func getAccessList(pagination: RequestPagination, syncDate: Int64?, completion: @escaping (Result<ResponseAccessPointList?, RequestError>) -> Void) {
        var url = "api/v2/listAccessPointsRequest?t=\(pagination.t ?? 100)&s=\(pagination.s ?? 0)"
        
        if (syncDate != nil) {
            url += "&d=\(syncDate!)"
        }
        
        BaseService.shared.requestGet(url: url, completion: completion)
    }
}

File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/service/DataService.java
package com.armongate.mobilepasssdk.service;

import com.armongate.mobilepasssdk.model.request.RequestPagination;
import com.armongate.mobilepasssdk.model.request.RequestSetUserData;
import com.armongate.mobilepasssdk.model.response.ResponseAccessPointList;

public class DataService {

    public void sendUserInfo(RequestSetUserData request, final BaseService.ServiceResultListener listener) {
        BaseService.getInstance().requestPost("api/v1/setpublickey", request, null, listener);
    }

    public void getAccessList(RequestPagination pagination, Long syncDate, final BaseService.ServiceResultListener<ResponseAccessPointList> listener) {
        String url = "api/v2/listAccessPointsRequest?t=" + pagination.t + "&s=" + pagination.s;

        if (syncDate != null) {
            url = url + "&d=" + syncDate.toString();
        }

        BaseService.getInstance().requestGet(url, ResponseAccessPointList.class, listener);
    }
}
